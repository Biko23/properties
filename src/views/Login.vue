<template>
  <div>
    <top-nav />
    <v-container id="main-container" fluid>
      <v-row>
        <v-col cols="12" sm="12" md="6" lg="6">
          <router-link
            to="/"
            style="
              color: black;
              text-align: center;
              margin-top: 40px;
              margin-left: 50px;
              text-decoration: none;
            "
          >
            <p class="back-home">
              <v-img
                style="margin-left: auto; margin-right: auto; display: block"
               
                max-height="30"
                max-width="28"
                src="https://res.cloudinary.com/diued7ugb/image/upload/v1626937941/Vector_9_aizknf.png"
              ></v-img
              ><span
                style="margin-left: 20px; margin-top: -20px; position: absolute"
              >
                Back home</span
              >
            </p>
          </router-link>
          <h1 id="get-started">Welcome Back!</h1>
          <p id="login-into">Login into your account</p>
        </v-col>
        <v-col>
          <v-form id="form">
            <div>
               <v-img
                  style="margin-left: 120px; margin-top: 10px;position:absolute;"
                  lazy-src="https://picsum.photos/id/11/10/6"
                  max-height="45"
                  max-width="45"
                  src="https://res.cloudinary.com/diued7ugb/image/upload/v1625749459/Vector_pebqf0.png"
                ></v-img>
              <span style="color: #3b6ef3; margin-top:23px;margin-left:-55px;position:absolute;">
                Stanbic Properties
              </span>
            </div><br><br>

            <v-container>
              <v-row>
                <v-col cols="12" sm="12" md="12">
                  <v-text-field
                    v-model="loginDetails.username"
                    label="Username or Email or phone number"
                    placeholder="Username or Email or phone number"
                    solo
                  ></v-text-field>
                </v-col>
              </v-row>
              <v-row>
                <v-col cols="12" sm="12" md="12">
                  <v-text-field
                    v-model="loginDetails.password"
                    label="Password"
                    placeholder="password"
                    type="password"
                    solo
                  ></v-text-field>
                </v-col>
              </v-row>
              <v-row>
                <v-col cols="12" sm="12" md="12">
                  <v-btn color="primary" @click="postLoginData" large block> Login </v-btn>
                </v-col> </v-row
              ><br /><br />
              <v-row>
                <v-divider></v-divider><span style="margin-top: -10px">Or</span>
                <v-divider></v-divider>
              </v-row>

              <v-row>
                <v-col cols="12" sm="6" md="6">
                  <v-btn color="#e7f0ff" block large>
                    <v-img
                      max-height="30"
                      max-width="20"
                      src="https://res.cloudinary.com/diued7ugb/image/upload/v1626938676/flat-color-icons_google_fsmdyf.png"
                    ></v-img>
                    <span style="padding-left: 5px"> Google</span>
                  </v-btn>
                </v-col>
                <v-col cols="12" sm="6" md="6">
                  <v-btn color="white" block large>
                    <v-img
                      max-height="34"
                      max-width="25"
                      src="https://res.cloudinary.com/diued7ugb/image/upload/v1626939162/Apple_logo_black.svg_rsk1qa.png"
                    >
                      
                    </v-img><span style="padding-left: 10px"></span>
                    Apple
                  </v-btn>
                </v-col> </v-row
              ><br />
              <p style="font-size: 0.75rem">
                Have No Accont?
                <router-link to="/signup">Sign up</router-link>
              </p>
              <br /><br />
            </v-container>
          </v-form>
        </v-col>
      </v-row>
    </v-container>
  </div>
</template>

<script>
import { mapActions } from 'vuex';
import TopNav from "../components/TopNav.vue";

export default {
  name: "Login",
  components: { TopNav },
  data: () => ({
    loginDetails: {
      username: '',
      password: ''
    }
  }),
  methods: {
    ...mapActions(["login", "fetchLoggedUser"]),
    async postLoginData (){
      try {
         const response = await this.login(this.loginDetails);
        if(response.status === 200 || response.status === 201){
          this.fetchLoggedUser()
            .then(()=>{
              this.$router.replace(sessionStorage.getItem('redirectPath') || '/');
              sessionStorage.removeItem('redirectPath');
          })
        }
      } catch (error) {
        throw new Error("Failed, Please try again");
      }
    }
  }
};
</script>

<style scoped>
#main-container {
  background: url("../assets/login.png");
  height: 100vh;
  background-size: cover;
}
#get-started {
  color: #3b6ef3;
  text-align: center;
  font-weight: 500;
  margin-top: 50px;
  margin-left: 70px;
  margin-right: 70x;
}
.back-home{
  margin-left: -100px; margin-top: 100px
}
#login-into {
  margin-top: 70px;
  text-align: center;
  margin-left: 90px;
}
#form {
  background-color: #e7f0ff;
  margin-left: auto;
  margin-right: auto;
  margin-top: 50px;
  width: 450px;
  border-radius: 6px;
  text-align: center;
}
@media only screen and (max-width: 768px) {
  #main-container {
    height: auto;
  }
  #form {
  background-color: #e7f0ff;
  margin-left: auto;
  margin-right: auto;
  margin-top: 50px;
  width: auto;
  border-radius: 6px;
  text-align: center;
}
#get-started {
  color: #3b6ef3;
  text-align: center;
  font-weight: 500;
  margin-top: auto;
  margin-left: auto;
  margin-right: auto;
}
#login-into {
  margin-top: auto;
  text-align: center;
  margin-left: auto;
}
.back-home{
  margin-left: -100px; margin-top: auto
}
}
</style>>

