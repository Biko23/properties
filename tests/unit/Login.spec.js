import { createLocalVue, mount } from '@vue/test-utils'
import Login from '@/views/Login.vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import VueRouter from 'vue-router'


describe('Login.vue', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VueRouter)
    let actions
    let getters
    let store
    let vuetify
    let router

    beforeEach(() => {
        vuetify = new Vuetify()
        router = new VueRouter()
        actions = {
            fetchLoggedUser: jest.fn(), 
            login: jest.fn(), 
            fetchTotalFavoriteCount: jest.fn(), 
            postAUserLog: jest.fn()        
        }
        store = new Vuex.Store({
            actions
        })
    })

    const mountFactory = options => {
        return mount(Login, {
            localVue,
            vuetify,
            store,
            router,
            ...options
        })
    }

    it('should have a match snapshot', () => {
        const wrapper = mountFactory({})
        expect(wrapper.html()).toMatchSnapshot()
    })
})

