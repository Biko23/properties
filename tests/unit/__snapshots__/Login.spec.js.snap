// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue should have a match snapshot 1`] = `
<div>
  <base-dialog message="" title=""></base-dialog>
  <div class="container container--fluid" id="main-container">
    <div class="row">
      <div class="col-sm-12 col-md-6 col-lg-6 col-12"><a href="#/" aria-current="page" class="router-link-exact-active router-link-active" style="color: black; text-align: center; margin-top: 40px; margin-left: 50px; text-decoration: none;">
          <p class="back-home">
            <div class="v-image v-responsive theme--light" style="max-height: 30px; max-width: 28px; margin-left: auto; margin-right: auto; display: block;">
              <transition-stub name="fade-transition" mode="in-out">
                <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
              </transition-stub>
              <div class="v-responsive__content"></div>
            </div><span style="margin-left: 20px; margin-top: -20px; position: absolute;">
                            Back home</span>
          </p>
        </a>
        <h1 id="get-started">Welcome Back!</h1>
        <p id="login-into">Login into your account</p>
      </div>
      <div class="col">
        <form novalidate="novalidate" id="form" class="v-form">
          <div>
            <div class="v-image v-responsive image-logo theme--light" style="max-height: 45px; max-width: 45px;">
              <transition-stub name="fade-transition" mode="in-out">
                <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
              </transition-stub>
              <div class="v-responsive__content"></div>
            </div> <span style="color: rgb(59, 110, 243); margin-top: 23px; margin-left: -55px; position: absolute;">
                            Stanbic Properties
                        </span>
          </div> <br><br>
          <div class="container">
            <div class="row">
              <div class="col-sm-12 col-md-12 col-12">
                <div class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--placeholder">
                  <div class="v-input__control">
                    <div class="v-input__slot">
                      <div class="v-text-field__slot"><label for="input-10" class="v-label theme--light" style="left: 0px; position: absolute;">Username or Email or phone number</label><input id="input-10" type="text"></div>
                    </div>
                    <div class="v-text-field__details">
                      <div class="v-messages theme--light">
                        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div class="row">
              <div class="col-sm-12 col-md-12 col-12">
                <div class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--placeholder">
                  <div class="v-input__control">
                    <div class="v-input__slot">
                      <div class="v-text-field__slot"><label for="input-13" class="v-label theme--light" style="left: 0px; position: absolute;">Password</label><input id="input-13" type="password"></div>
                      <div class="v-input__append-inner">
                        <div class="v-input__icon v-input__icon--append"><button type="button" aria-label="append icon" class="v-icon notranslate v-icon--link mdi mdi-eye-off theme--light"></button></div>
                      </div>
                    </div>
                    <div class="v-text-field__details">
                      <div class="v-messages theme--light">
                        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div class="row">
              <div class="col-sm-12 col-md-12 col-12"><button type="button" class="v-btn v-btn--block v-btn--is-elevated v-btn--has-bg theme--light v-size--large primary"><span class="v-btn__content">
                                    Login
                                </span></button></div>
            </div> <span style="color: red;"></span> <br><br> <br>
            <p style="font-size: 0.85rem;">
              Have No Accont?
              <a href="#/signup" class="">Sign up</a>
            </p>
            <p style="font-size: 0.75rem;"><a href="#/email-reset-password" class="">Forgot Password</a></p> <br><br>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
